pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -441,7 +444,15 @@ private InstanceResponseBlock executeInternal(TableDataManager tableDataManager,
       if (queryContext.isExplain()) {
         instanceResponse = getExplainResponseForNoMatchingSegment(numTotalSegments, queryContext);
       } else {
-        instanceResponse = new InstanceResponseBlock(ResultsBlockUtils.buildEmptyQueryResults(queryContext));
+        if (QueryContextUtils.isTimeSeriesQuery(queryContext)) {
+          // TODO: handle invalid segments
```

Comment:
What will happen now if we get invalid segments? In what cases invalid segments can reach here?